lte/gateway/c/oai/tasks/amf/amf_asDefs.h

```diff
@@ -0,0 +1,173 @@
+/**
+ * Copyright 2020 The Magma Authors.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+#include <sstream>
+#ifdef __cplusplus
+extern "C" {
+#endif
+#include "3gpp_24.008.h"
+#ifdef __cplusplus
+};
+#endif
+#include "TrackingAreaIdentity.h"
+#include "common_types.h"
+#include "amf_config.h"
+#include "amf_securityDef.h"
+
+#define AUTH_KNAS_INT_SIZE 16 /* NAS integrity key     */
+#define AUTH_KNAS_ENC_SIZE 16 /* NAS cyphering key     */
+
+namespace magma5g {
+// Data used to setup 5g CN NAS security /
+typedef struct amf_as_security_data_s {
+  bool is_new;                           // New security data indicator
+  ksi_t ksi;                             // NAS key set identifier
+  uint8_t sqn;                           // Sequence number
+  uint32_t count;                        // NAS counter
+  uint8_t knas_enc[AUTH_KNAS_ENC_SIZE];  // NAS cyphering key
+  uint8_t knas_int[AUTH_KNAS_INT_SIZE];  // NAS integrity key
+  bool is_knas_enc_present;
+  bool is_knas_int_present;
+} amf_as_security_data_t;
+
+// AMFAS-SAP primitives
+typedef enum amf_as_primitive_s {
+  _AMFAS_START = 200,
+  _AMFAS_SECURITY_REQ,   // AMF->AS: Security request
+  _AMFAS_SECURITY_IND,   // AS->AMF: Security indication
+  _AMFAS_SECURITY_RES,   // AMF->AS: Security response
+  _AMFAS_SECURITY_REJ,   // AMF->AS: Security reject
+  _AMFAS_ESTABLISH_REQ,  // AMF->AS: Connection establish request
+  _AMFAS_ESTABLISH_CNF,  // AS->AMF: Connection establish confirm
+  _AMFAS_ESTABLISH_REJ,  // AS->AMF: Connection establish reject
+  _AMFAS_RELEASE_REQ,    // AMF->AS: Connection release request
+  _AMFAS_RELEASE_IND,    // AS->AMF: Connection release indication
+  _AMFAS_DATA_REQ,       // AMF->AS: Data transfer request
+  _AMFAS_DATA_IND,       // AS->AMF: Data transfer indication
+  _AMFAS_PAGE_IND,       // AS->AMF: Paging data indication
+  _AMFAS_STATUS_IND,     // AS->AMF: Status indication
+  _AMFAS_END
+} amf_as_primitive_t;
+
+typedef enum amfcn_primitive_s {
+  _AMFCN_START = 400,
+  AMFCN_AUTHENTICATION_PARAM_RES,
+  AMFCN_AUTHENTICATION_PARAM_FAIL,
+  AMFCN_NW_INITIATED_DEREGISTRATION_UE,
+  AMFCN_DEACTIVATE_PDUSESSION_REQ,
+  AMFCN_IDENTITY_PARAM_RES,
+  AMFCN_SMC_PARAM_RES,
+  _AMFCN_END
```

Comment:
remove _ at the beginning. Please check any usage of `_` or `__` at the beginning of variables, object or function names and remove them.

Reply:
Those enums are leveraged from LTE, we followed the same naming. Remaining variables, objects and functions names are taken care.

Leveraged enums are used in many places if we need to remove the _ at the begining of those, we will address them in separate PR.